### ISSUE TITLE
Magento 2.2 checkout error

### ISSUE THREAD

**Issue by glivanis** [VivaPayments/API#1547]
Plugin installed and enabled on Magento 2.2.9 / 

- API keys are enabled OK and approved
- OrderCode URL is https://vivapayments.com/api/orders
- Gateway is https://vivapayments.com/web/newtransaction.aspx
- Transaction URL is https://vivapayments.com/api/transactions/
but at the checkout process raises an error at  **/vivapayments/checkout/start** page

> There has been an error processing your request

Viva Wallet was installed and working OK on Magento 1.9. Then it was migrated to Magento 2. Old plugin was deleted and installed the latest CED plugin for Magento 2.2 (Maybe some database tables should be removed manually ???)

Magento Error log on the attached txt file:

[errorlog.txt](https://github.com/VivaPayments/API/files/4431120/errorlog.txt)

**Comment by matthewrgourd** [comment]
Hi @glivanis did you follow the instructions on our [Magento 2.10-2.2 page](https://developer.vivawallet.com/e-commerce-plugins/magento2.1-2.2/) including using the download link for the plugin from the top?

**Comment by glivanis** [comment]
Hi there, thank you for your reply.
Yes, I did follow the instructions, actually the plugin was working OK on my previous installation (Magento 1.9). I had the plugin installed and re-installed - deployed a couple of times and I've noticed that the settings are coming back by themselves, so there are left overs on the database. Is there a safe way to **completely** erase files and settings and clean re-install? Is there a chance that data is stored in other folder that the CED folder? Attached are the requested files.
Thank you in advance

![screenshot1](https://user-images.githubusercontent.com/37864668/78984572-446e9a80-7b2f-11ea-93a3-ded4ced1e1b8.jpg)
![screenshot3](https://user-images.githubusercontent.com/37864668/78984578-459fc780-7b2f-11ea-8024-3185c650ad17.jpg)
![screenshot2](https://user-images.githubusercontent.com/37864668/78984580-46d0f480-7b2f-11ea-8a79-e838e76da94a.jpg)
![screenshot4](https://user-images.githubusercontent.com/37864668/78984581-47698b00-7b2f-11ea-888a-78c93b675c8f.jpg)

**Comment by matthewrgourd** [comment]
In answer to your last question, I'm not a Magento expert, I'm afraid, so not sure. However, in response to your generic installation issue, please check again our plugin installation guide as we updated it recently:

https://developer.vivawallet.com/e-commerce-plugins/magento2.1-2.2/

Please see in particular step 8:
In the Magento store settings:

MerchantID – your Viva Wallet Merchant ID
API Key – your Viva Wallet API Key
Source Code – 4-digit code for the payment source you set up in the Viva Wallet banking app
OrderCode URL – https://www.vivapayments.com/api/orders
Gateway URL – https://www.vivapayments.com/web/newtransaction.aspx
Transaction URL – https://www.vivapayments.com/api/transactions

**Comment by glivanis** [comment]
To anyone it might concern, I fixed my problem by following these steps: 
- **Backed up everything**, just in case. 
- I Uninstalled the plugin with the composer and Deleted the VivaPayments folder from the server
- **Edited the database** and manualy deleted any reference to VivaPayments
- Re-installed the plugin according to instructions
- Manually deleted the contents of the /generated and var/view_preprocessed folders, apart from .htaccess files.
- **Only Then** I did the update - compile - deploy - cache flush sequence.
Looks like the migration from Magento 1.9x to 2x is causing some trouble.
